langchain4j-core/src/main/java/dev/langchain4j/model/input/MustachePromptTemplateFactory.java

```diff
@@ -0,0 +1,35 @@
+package dev.langchain4j.model.input;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import dev.langchain4j.spi.prompt.PromptTemplateFactory;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.Map;
+
+class MustachePromptTemplateFactory implements PromptTemplateFactory {
+
+    private static final MustacheFactory MUSTACHE_FACTORY = new DefaultMustacheFactory();
+
+    @Override
+    public MustacheTemplate create(PromptTemplateFactory.Input input) {
```

Comment:
Could you please explain the purpose of interface `Input`? Why not just passing `String template` as an argument to the factory?

Reply:
Sure!

The reason I have this is because there are actually two things required, the `template` and the `name`. 
Experience has shown us, that when there are multiple parameters, they have a way to multiple in time when more integrations come along. If that happens, having such an interface allows for seamless introduction of the new parameter, without having to either break existing SPI impls or add odd default methods